File: src/main/java/me/derechtepilz/economy/utility/APIRequest.java
package me.derechtepilz.economy.utility;

import me.derechtepilz.economy.utility.exceptions.UnsuccessfulAPIRequestException;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class APIRequest {
    private HttpURLConnection connection;
    private String line;
    private BufferedReader reader;
    private StringBuffer responseContent = new StringBuffer();

    public String request(String url) throws IOException {
        URL requestUrl = new URL(url);
        connection = (HttpURLConnection) new URL(url).openConnection();
        connection.setRequestMethod("GET");
        int responseCode = connection.getResponseCode();
        if (responseCode > 299) {
            throw new UnsuccessfulAPIRequestException("Could not finish api request!");
        } else {
            reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
        }
        while ((line = reader.readLine()) != null) {
            responseContent.append(line);
        }
        reader.close();
        return responseContent.toString();
    }
}

File: economy-plugin/src/main/kotlin/io/github/derechtepilz/economy/updatemanagement/APIRequest.kt
package io.github.derechtepilz.economy.updatemanagement

import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class APIRequest(private val url: String) {
	private var reader: BufferedReader? = null

	fun request(): String {
		val connection: HttpURLConnection = URL(url).openConnection() as HttpURLConnection
		connection.requestMethod = "GET"
		val responseCode: Int = connection.responseCode
		if (responseCode > 299) {
			throw IllegalStateException("Could not finish api request!")
		} else {
			reader = BufferedReader(InputStreamReader(connection.inputStream))
		}
		val responseContent = StringBuffer()
		var line = reader!!.readLine()
		while (line != null) {
			responseContent.append(line)
			line = reader!!.readLine()
		}
		reader!!.close()
		return responseContent.toString()
	}
}